### PR TITLE
fix: end at field for video

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

EndAt field on Admin currently does nothing. It is expected to go to the next step when it reaches the EndAt time. And if the video is muted it should loop the video from EndAt to Start At

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/27244374/todos/4874291938#__recording_4876692575)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] For background videos only - if the video has an EndAt value it should loop (from EndAt to StartAt) and not change the card
- [ ] If a video has an endAt value it should pause at EndAt and jump to the next card.
- [ ] The assigned action to a video should occur as expected

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/VideoBlockEditorSettings.tsx [hunk 1]
```diff
@@ -153,6 +153,7 @@ export function VideoBlockEditorSettings({
           />
           <TimeField
             showSeconds
+            value={values.endAt}
```
[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/VideoBlockEditorSettings.tsx:156]: I found that once the EndAt field gets a value (user inputs time on admin) it doesn't update the front end visually but it updates the database. And it just reverts back to what you've set to be all zeros (00:00:00). Adding that prop seems to fix that problem

File: libs/journeys/ui/src/components/Card/Cover/Cover.tsx [hunk 2]
```diff
@@ -65,6 +65,21 @@ export function Cover({
...
+          videoBlock?.endAt != null &&
+          videoBlock?.endAt > 0 &&
+          playerRef.current?.currentTime() != null
+        ) {
+          if (
+            playerRef.current?.currentTime() < videoBlock?.startAt ||
+            playerRef.current?.currentTime() >= videoBlock?.endAt
+          ) {
+            playerRef.current?.currentTime(videoBlock?.startAt ?? 0)
+          }
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Card/Cover/Cover.tsx:80]:
```suggestion
	const currentTime = playerRef.current.currentTime()
	const { startAt, endAt } = videoBlock
          if (
            currentTime < startAt ?? 0 || currentTime >= endAt
          ) {
            playerRef.current.currentTime(startAt ?? 0)
          }
```

File: libs/journeys/ui/src/components/Video/VideoTrigger/VideoTrigger.tsx [hunk 3]
```diff
@@ -33,7 +36,7 @@ export function VideoTrigger({
       player.on('timeupdate', timeUpdate)
       return () => player.off('timeupdate', timeUpdate)
     }
-  }, [player, triggerStart, router, triggerAction, triggered])
+  }, [player, router, triggered, triggerAction, triggerStart])
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Video/VideoTrigger/VideoTrigger.tsx:39]: revert?

**Review by tataihono — CHANGES_REQUESTED** [review]
revert and code simplification

**Review by tataihono — APPROVED** [review]

**Pull request merged**